universal-login-react/src/ui/Onboarding/ConfirmCode.tsx

```diff
@@ -26,14 +29,17 @@ const getSubText = (isValid: undefined | boolean) => {
   }
 };
 
-export const ConfirmCode = ({email}: ConfirmCodeProps) => {
+export const ConfirmCode = ({email, onConfirmCode, walletService}: ConfirmCodeProps) => {
   const [code, setCode] = useState<string | undefined>(undefined);
   const [isValid, setIsValid] = useState<boolean | undefined>(undefined);
 
-  const onConfirmClick = () => {
+  const onConfirmClick = async () => {
     ensureNotFalsy(code, Error, 'Code is missing');
     ensure(code?.length === CODE_LENGTH, Error, 'Code is incomplete.');
+    await walletService.confirmCode(code);
     setIsValid(true);
+    await sleep(1000);
```

Comment:
What is purpose for `sleep`?

Reply:
We have success component, so the user can see it